eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -1209,6 +1209,43 @@ public function testSwapLocation()
         );
     }
 
+    /**
+     * Test swapping Main Location of a Content with another one updates Content item Main Location.
+     *
+     * @covers \eZ\Publish\API\Repository\LocationService::swapLocation
+     */
+    public function testSwapLocationUpdatesMainLocation()
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+        $contentService = $repository->getContentService();
+
+        $mainLocationParentId = 60;
+        $secondaryLocationId = 43;
+
+        $publishedContent = $this->publishContentWithParentLocation(
+            'Content for Swap Location Test', $mainLocationParentId
+        );
+
+        // sanity check
+        $mainLocation = $locationService->loadLocation($publishedContent->contentInfo->mainLocationId);
+        self::assertEquals($mainLocationParentId, $mainLocation->parentLocationId);
+
+        // load another pre-existing location
+        $secondaryLocation = $locationService->loadLocation($secondaryLocationId);
+
+        // swap the Main Location with a secondary one
+        $locationService->swapLocation($mainLocation, $secondaryLocation);
+
+        // check if Main Location has been updated
+        $mainLocation = $locationService->loadLocation($secondaryLocation->id);
+        self::assertEquals($publishedContent->contentInfo->id, $mainLocation->contentInfo->id);
+        self::assertEquals($mainLocation->id, $mainLocation->contentInfo->mainLocationId);
```

Comment:
Note: @andrerom This assert and the next one fails if we don't invalidate `location-<locationId>` tag on swap also.